File: desafioPartidasRankeadas.py
nome_heroi = "Rei Arthur"

def saldo_Rankeadas(vitorias, derrotas):
    saldo_Vitorias = vitorias - derrotas
    return saldo_Vitorias

def determinar_Nivel(saldo_Vitorias):
    niveis = {
        range(0, 11): "Ferro",
        range(11, 21): "Bronze",
        range(21, 51): "Prata",
        range(51, 81): "Ouro",
        range(81, 91): "Diamante",
        range(91, 101): "Lendário",
        range(101, 300): "Imortal"
    }

    for faixa, nivel in niveis.items():
        if saldo_Vitorias in faixa:
            return nivel

def main():
    saldo = saldo_Rankeadas(200, 100)
    nivel = determinar_Nivel(saldo)
    print("O Herói", nome_heroi, "tem o saldo de", saldo, "e está no nível", nivel)

main()

File: desafioPartidasRankeadas.js
let nomeheroi = "Rei Arthur"
let nivelheroi

saldoRankeadas(200,100)

function saldoRankeadas (vitorias , derrotas){
    saldoVitorias = vitorias - derrotas
    return saldoVitorias
}

if (saldoVitorias <= 10){
    nivelheroi = "Ferro"
} else if (saldoVitorias >= 11 && saldoVitorias <= 20){
    nivelheroi = "Bronze"
} else if (saldoVitorias >= 21 && saldoVitorias <= 50){
    nivelheroi = "Prata"
} else if (saldoVitorias >= 51 && saldoVitorias <= 80){
    nivelheroi = "Ouro"
} else if (saldoVitorias >= 81 && saldoVitorias <= 90){
    nivelheroi = "Diamante"
} else if (saldoVitorias >= 91 && saldoVitorias <= 100){
    nivelheroi = "Lendário"
} else if (saldoVitorias >= 101){
    nivelheroi = "Imortal"
}

console.log( "O Herói " + nomeheroi + " tem o saldo de " + saldoVitorias + " e está no nível " + nivelheroi)
